consult.el

```diff
@@ -4587,6 +4587,23 @@ If NORECORD is non-nil, do not record the buffer switch in the buffer list."
                                        :as #'consult--buffer-pair)))
   "Buffer candidate source for `consult-buffer'.")
 
+(defvar consult--source-tab-buffer
+  `(:name     "Tab buffer"
+    :narrow   ?b
+    :category buffer
+    :face     consult-buffer
+    :history  buffer-name-history
+    :state    ,#'consult--buffer-state
+    :default  t
+    :items
+    ,(lambda ()
+       (consult--buffer-query
+        :sort 'visibility
+        :as #'buffer-name
+        :predicate #'(lambda (buf)
+                       (memq buf (frame-parameter nil 'buffer-list))))))
```

Comment:
The predicate looks costy. Maybe let-bind the variable around the lambda? Also the #' is unnecessary.

Reply:
> Maybe let-bind the variable around the lambda?

I don't understand what I should do here.

Reply:
I meant `(let ((bufs (frame-parameter ...)) (lambda (buf) (memq buf bufs)))`.